### PR TITLE
Convert to edd_get_option

### PR DESCRIPTION
**Pull request by ghost1227**
Update PR for 2.3 Issue #2429

### CODE REVIEW

**Comment by pippinsplugins** [comment]
I thought we already did all of this for the Polyglot issue. Are there left overs?

**Comment by ghost1227** [comment]
Yep, lots. At least in release/2.3...

**Comment by pippinsplugins** [comment]
This should be unneeded now #3011 but we'll leave open to confirm.

File: includes/class-edd-cache-helper.php [hunk 1]
```diff
@@ -32,21 +32,19 @@ public function init() {
 
 		if ( false === ( $page_uris = get_transient( 'edd_cache_excluded_uris' ) ) ) {
 
-			global $edd_options;
-
-			if( empty( $edd_options['purchase_page'] ) || empty( $edd_options['success_page'] ) ) {
+			if( ! edd_get_option( 'purchase_page', false ) || ! edd_get_option( 'success_page' ) ) {
```
Comment (cklosowski) [inline includes/class-edd-cache-helper.php]: Are we missing a default option value for the ```! edd_get_option( 'success_page' )``` check here?

File: includes/admin/downloads/dashboard-columns.php [hunk 2]
```diff
@@ -49,12 +49,10 @@ function edd_download_columns( $download_columns ) {
...
-		global $edd_options;
-
-		$style 			= isset( $edd_options['button_style'] ) ? $edd_options['button_style'] : 'button';
-		$color 			= isset( $edd_options['checkout_color'] ) ? $edd_options['checkout_color'] : 'blue';
-		$color			= ( $color == 'inherit' ) ? '' : $color;
-		$purchase_text 	= ! empty( $edd_options['add_to_cart_text'] ) ? $edd_options['add_to_cart_text'] : __( 'Purchase', 'edd' );
+		$style			= edd_get_option( 'button_style', 'button' );
+		$color			= edd_get_option( 'checkout_color', 'blue' );
+		$color			= ( $color == 'inherit' ) ? '' : $color;		
+		$purchase_text  = edd_get_option( 'add_to_cart_text', __( 'Purchase', 'edd' ) );
```
Comment (cklosowski) [inline includes/admin/downloads/dashboard-columns.php]: Clean up the white spacing here, aligning the ```=```

File: includes/gateways/paypal-standard.php [hunk 3]
```diff
@@ -66,14 +63,14 @@ function edd_process_paypal_purchase( $purchase_data ) {
...
 
 		// Get the PayPal redirect uri
 		$paypal_redirect = trailingslashit( edd_get_paypal_redirect() ) . '?';
 
 		// Setup PayPal arguments
 		$paypal_args = array(
-			'business'      => $edd_options['paypal_email'],
+			'business'      => edd_get_option( 'paypal_email' ),
 			'email'         => $purchase_data['user_email'],
 			'invoice'  => $purchase_data['purchase_key'],
```
Comment (cklosowski) [inline includes/gateways/paypal-standard.php]: While we're in here, can you update these ```=>``` to align?

File: includes/shortcodes.php [hunk 4]
```diff
@@ -33,9 +33,9 @@ function edd_download_shortcode( $atts, $content = null ) {
 		'sku'			=> '',
 		'price'         => '1',
 		'direct'        => '0',
-		'text'	        => isset( $edd_options[ 'add_to_cart_text' ] )  && $edd_options[ 'add_to_cart_text' ]    != '' ? $edd_options[ 'add_to_cart_text' ] : __( 'Purchase', 'edd' ),
-		'style'         => isset( $edd_options[ 'button_style' ] ) 	 	? $edd_options[ 'button_style' ] 		: 'button',
-		'color'         => isset( $edd_options[ 'checkout_color' ] ) 	? $edd_options[ 'checkout_color' ] 		: 'blue',
+		'text'	=> edd_get_option( 'add_to_cart_text', __( 'Purchase', 'edd' ) ),
+		'style' => edd_get_option( 'button_style', 'button' ),
+		'color' => edd_get_option( 'checkout_color', 'blue' ),
```
Comment (cklosowski) [inline includes/shortcodes.php]: White spacing clean up for these lines, just make sure spaces are used between the ```key``` and ```=>```

File: includes/shortcodes.php [hunk 5]
```diff
@@ -228,14 +228,12 @@ function edd_discounts_shortcode( $atts, $content = null ) {
...
-	global $edd_options;
-
 	extract( shortcode_atts( array(
 			'taxonomy'	=> '',
 			'terms'		=> '',
 			'text'		=> __('Purchase All Items', 'edd'),
-			'style'		=> isset( $edd_options['button_style'] ) ? $edd_options['button_style'] : 'button',
-			'color'		=> isset( $edd_options['checkout_color'] ) ? $edd_options['checkout_color'] : 'blue',
+			'style'	=> edd_get_option( 'button_style', 'button' ),
+			'color'	=> edd_get_option( 'checkout_color', 'blue' ),
```
Comment (cklosowski) [inline includes/shortcodes.php]: More whitespace cleanup please.

File: includes/template-functions.php [hunk 6]
```diff
@@ -62,9 +62,9 @@ function edd_get_purchase_link( $args = array() ) {
 		'price'       => (bool) true,
 		'price_id'    => isset( $args['price_id'] ) ? $args['price_id'] : false,
 		'direct'      => edd_get_download_button_behavior( $post_id ) == 'direct' ? true : false,
-		'text'        => ! empty( $edd_options[ 'add_to_cart_text' ] ) ? $edd_options[ 'add_to_cart_text' ] : __( 'Purchase', 'edd' ),
-		'style'       => isset( $edd_options[ 'button_style' ] ) 	   ? $edd_options[ 'button_style' ]     : 'button',
-		'color'       => isset( $edd_options[ 'checkout_color' ] ) 	   ? $edd_options[ 'checkout_color' ] 	: 'blue',
+		'text' => edd_get_option( 'add_to_cart_text', __( 'Purchase', 'edd' ) ),
+		'style' => edd_get_option( 'button_style', 'button' ),
+		'color' => edd_get_option( 'checkout_color', 'blue' ),
```
Comment (cklosowski) [inline includes/template-functions.php]: White space here too.

**Comment by cklosowski** [comment]
Ok @ghost1227 left some notes inline about some changes. Mostly just white space, one question on if we should include a default value on one of the calls.

File: includes/class-edd-cache-helper.php (hunk 1 above)
Comment (ghost1227) [inline includes/class-edd-cache-helper.php]: No. The original didn't have a default, and we can't predict what pages will or will not exist with any certainty. If we assume a page exists and it doesn't (or isn't published) things don't work right.

Comment (cklosowski) [inline includes/class-edd-cache-helper.php]: But we did it in the first check:
```edd_get_option( 'purchase_page', false )```

Comment (ghost1227) [inline includes/class-edd-cache-helper.php]: That's valid. Technically, the current expression returns as false so the end result is the same though.

**Comment by ghost1227** [comment]
@cklosowski finally, build passing and cleanly merges

**Comment by cklosowski** [comment]
Cool, will wait for CI to finish and take a peek.

**Pull request merged**